src/pkg/cli/client/byoc/aws/byoc.go

```diff
@@ -107,14 +115,24 @@ func AnnotateAwsError(err error) error {
 	return err
 }
 
-func NewByocProvider(ctx context.Context, tenantId types.TenantID) *ByocAws {
+type NewByocInterface func(ctx context.Context, tenantId types.TenantID) *ByocAws
+
+func newByocProvider(ctx context.Context, tenantId types.TenantID) *ByocAws {
 	b := &ByocAws{
 		driver: cfn.New(byoc.CdTaskPrefix, aws.Region("")), // default region
 	}
 	b.ByocBaseClient = byoc.NewByocBaseClient(ctx, tenantId, b)
+
+	if StsClient == nil {
+		cfg := aws.PanicOnError(b.driver.LoadConfig(ctx))
+		StsClient = sts.NewFromConfig(cfg)
```

Comment:
Not liking the `panic`. What would it look like if we'd add the `stsClient` pointer to the `ByocAws` structure? And change the old code to "create if not initialized"?